src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback } from 'react';
+import Button from '@material-ui/core/Button';
+import DeleteIcon from '@material-ui/icons/Delete';
+import FormControl from '@material-ui/core/FormControl';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+
+import { makeStyles } from '@material-ui/core/styles';
+import { useForm } from 'react-final-form';
+
+const useStyles = makeStyles({
+  row: {
+    zIndex: 10,
+    margin: '1em 0',
+    backgroundColor: 'white',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  formControl: {
+    display: 'flex',
+    flexDirection: 'columns',
+    alignItems: 'strech',
+  },
+});
+
+const PopupFieldRow = ({ popupField, onChange }) => {
+  const classes = useStyles();
+
+  const onSuffixChange = useCallback(e => {
+    const newPopupField = { ...popupField };
+    newPopupField.suffix = e.target.value;
+    onChange(newPopupField);
+  }, [onChange, popupField]);
+
+  const onPrefixChange = useCallback(e => {
+    const newPopupField = { ...popupField };
+    newPopupField.prefix = e.target.value;
+    onChange(newPopupField);
+  }, [onChange, popupField]);
+
+  const onDefaultChange = useCallback(e => {
+    const newPopupField = { ...popupField };
+    newPopupField.default = e.target.value;
+    onChange(newPopupField);
+  }, [onChange, popupField]);
+
+  const onDeletePopupField = useCallback(() => {
+    const newPopupField = { ...popupField };
+    newPopupField.deleted = true;
+    onChange(newPopupField);
+  }, [onChange, popupField]);
+
+  return (
+    <Paper className={classes.row}>
+      <FormControl className={classes.formControl}>
+        <div>{popupField.field}</div>
+      </FormControl>
+      <FormControl>
+        <TextField
+          label="prefix"
+          onChange={onPrefixChange}
+          value={popupField.prefix}
+          required
+        />
+      </FormControl>
+      <FormControl>
+        <TextField
+          label="suffix"
+          onChange={onSuffixChange}
+          value={popupField.suffix}
+          required
+        />
+      </FormControl>
+      <FormControl>
+        <TextField
+          label="default"
+          onChange={onDefaultChange}
+          value={popupField.default}
+          required
+        />
+      </FormControl>
+      <Button type="button" onClick={onDeletePopupField}><DeleteIcon /></Button>
+    </Paper>
+  );
+};
+
+const PopupFieldList = ({ fields, popupFields }) => {
+  const form = useForm();
+  const onChange = useCallback(popupField => {
+    const popupConfig = form.getState().values.popup_config;
+    const { wizard = {} } = popupConfig;
+    const { fields: wizardFields = [] } = wizard;
```

Comment:
I think you can do something like that:
```suggestion
    const { wizard: { fields: wizardFields = [] } = {}, wizard } = popupConfig;
```

Reply:
```suggestion
    const {
      wizard: { fields: wizardFields = [] } = {},
      wizard,
    } = popupConfig;
```

Reply:
Fantastic!